<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>768</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>微软雅黑</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>文法问题处理器 BNFProcessor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_console">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>240</y>
      <width>321</width>
      <height>461</height>
     </rect>
    </property>
    <property name="title">
     <string>控制台</string>
    </property>
    <widget class="QPlainTextEdit" name="plainTextEdit_console">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>301</width>
       <height>371</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Courier New</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clearConsole">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>420</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>清空</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_edit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>891</width>
      <height>691</height>
     </rect>
    </property>
    <property name="title">
     <string>操作台</string>
    </property>
    <widget class="QTabWidget" name="tabWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>871</width>
       <height>641</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>7</number>
     </property>
     <widget class="QWidget" name="tab_edit">
      <attribute name="title">
       <string>文法编辑</string>
      </attribute>
      <widget class="QPlainTextEdit" name="plainTextEdit_edit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>731</width>
         <height>541</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_arrowChar">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>10</y>
         <width>51</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>→</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_orChar">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>10</y>
         <width>51</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>|</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_epsilonChar">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>10</y>
         <width>51</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>ɛ</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>14</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>快捷编辑：</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_process">
       <property name="geometry">
        <rect>
         <x>750</x>
         <y>290</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>全部处理</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_simplified">
      <attribute name="title">
       <string>文法化简</string>
      </attribute>
      <widget class="QPlainTextEdit" name="plainTextEdit_simplified">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_check">
      <attribute name="title">
       <string>消除左递归</string>
      </attribute>
      <widget class="QPlainTextEdit" name="plainTextEdit_leftRecursion">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>提取左公因子</string>
      </attribute>
      <widget class="QPlainTextEdit" name="plainTextEdit_leftCommonFactor">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_first">
      <attribute name="title">
       <string>First集</string>
      </attribute>
      <widget class="QTableWidget" name="tableWidget_firstSet">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <column>
        <property name="text">
         <string>符号</string>
        </property>
        <property name="font">
         <font>
          <family>JetBrains Mono</family>
          <pointsize>10</pointsize>
         </font>
        </property>
       </column>
       <column>
        <property name="text">
         <string>集合</string>
        </property>
        <property name="font">
         <font>
          <family>JetBrains Mono</family>
          <pointsize>10</pointsize>
         </font>
        </property>
       </column>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Follow集</string>
      </attribute>
      <widget class="QTableWidget" name="tableWidget_followSet">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <column>
        <property name="text">
         <string>符号</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>集合</string>
        </property>
       </column>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_LL1">
      <attribute name="title">
       <string>LL(1)分析表</string>
      </attribute>
      <widget class="QTableWidget" name="tableWidget_LL1">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>851</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_CST">
      <attribute name="title">
       <string>语法树</string>
      </attribute>
      <widget class="QTreeWidget" name="treeWidget_CST">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>10</y>
         <width>501</width>
         <height>591</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Courier New</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
      <widget class="QPlainTextEdit" name="plainTextEdit_CST">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>341</width>
         <height>551</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>JetBrains Mono NL</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="placeholderText">
        <string>在此处输入需要语法分析的源程序</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_language">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>570</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>TINY</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>MiniC</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>570</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>源程序语言：</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>10</y>
      <width>171</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>操作3-6</string>
    </property>
    <widget class="QPushButton" name="pushButton_eliminateLeftCommonFactor">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>提取左公因子</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_set">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>集合求解</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_LL1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>生成LL(1)分析表</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_CST">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>语法分析</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clearAll">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>190</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>复位</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>10</y>
      <width>131</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>文法</string>
    </property>
    <widget class="QPushButton" name="pushButton_save">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>保存</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_open">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>打开</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>120</y>
      <width>131</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>操作1-2</string>
    </property>
    <widget class="QPushButton" name="pushButton_simplify">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>文法化简</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_eliminateLeftRecursion">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>消除左递归</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_edit">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="action_simplify"/>
    <addaction name="action_check"/>
    <addaction name="action_set"/>
   </widget>
   <widget class="QMenu" name="menu_H">
    <property name="title">
     <string>帮助(&amp;H)</string>
    </property>
    <addaction name="actionAbout_me"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_edit"/>
   <addaction name="menu_H"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_open">
   <property name="text">
    <string>打开(&amp;O)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>保存(&amp;S)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_simplify">
   <property name="text">
    <string>文法化简</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_check">
   <property name="text">
    <string>文法检查</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+C</string>
   </property>
  </action>
  <action name="action_set">
   <property name="text">
    <string>集合求解</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+S</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>退出(&amp;Q)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Q</string>
   </property>
  </action>
  <action name="actionAbout_me">
   <property name="text">
    <string>About me.😋</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
